src/options/coerce_custom.ts

```diff
@@ -0,0 +1,239 @@
+import {
+  ReadPreferenceMode,
+  Compressor,
+  ReadConcernLevel,
+  AuthMechanism,
+  LogLevel,
+  UriOptions,
+  ClientOptions
+} from './types';
+import { Coerce as C, CoerceMatch, CoerceOptions } from './coerce';
+
+export class CoerceCustom {
+  // prettier-ignore
```

Comment:
This may be a point of contention, but I'd prefer to ignore some of these blocks for our own sanity.

If not ok the alternative is this:

```ts
  static hedgeDef = (match: CoerceMatch) => ({
    ...match('enable', C.require(C.boolean))
  });

  static hedge = C.objectExact(CoerceCustom.hedgeDef);

  static readPreferenceDef = (match: CoerceMatch) => ({
    ...match('mode', C.default(C.enum({ ReadPreferenceMode }), 'primary')),
    ...match('maxStalenessSeconds', C.number),
    ...match('tags', C.default(C.tags, [])),
    ...match('hedge', C.objectExact())
  });

  static readPreference = C.warn(C.objectExact(CoerceCustom.readPreferenceDef));
```